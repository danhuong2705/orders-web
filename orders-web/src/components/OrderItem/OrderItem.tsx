import styled from "@emotion/styled";
import React from "react";
import { ORDER_STATE } from "../../contansts";
import { orderStateMapping } from "../../helpers";
import { IOrderLine } from "../../interface/OrderLine";

interface OrderItemProps {
  orderId: string;
  createdAt: string;
  orderItems: Array<IOrderLine>;
  state: ORDER_STATE;
  onViewOrderDetail: (orderId: string) => void;
}
interface StyledOrderItemProps {
  state: ORDER_STATE;
}
const Wrapper = styled.div`
  display: flex;
  justify-content: flex-start;
  padding: 1rem 0 1rem 0;
  cursor: pointer;
  border-bottom: 1px solid #e0e0e0;
  :hover {
    background-color: #e0e0e0;
  }
`;

const DateTime = styled.div``;
const Info = styled.div`
  display: flex;
  justify-content: space-between;
  margin-left: 2rem;
  margin-right: 2rem;
  flex: 1;
`;

const State = styled.div<StyledOrderItemProps>(
  ({ state }: StyledOrderItemProps) => `
  color: ${
    state === ORDER_STATE.CONFIRMED
      ? "#dede01"
      : state === ORDER_STATE.DELIVERED
      ? "#71e463"
      : state === ORDER_STATE.CANCELLED
      ? "#de4444"
      : "#000000"
  };
  font-weight: 500;
  `
);
const OrderItem: React.FC<OrderItemProps> = ({
  orderId,
  orderItems,
  createdAt,
  state,
  onViewOrderDetail,
}: OrderItemProps) => {
  return (
    <Wrapper onClick={() => onViewOrderDetail(orderId)}>
      <DateTime>{createdAt.substring(0, 10)}</DateTime>
      <Info>
        {orderItems[0].name}{" "}
        {orderItems.length > 1 &&
          `and ${orderItems.length - 1} other product(s)`}{" "}
      </Info>
      <State state={state}>{orderStateMapping(state)}</State>
    </Wrapper>
  );
};

export default OrderItem;
