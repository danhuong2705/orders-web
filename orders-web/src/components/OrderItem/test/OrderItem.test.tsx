import React from "react";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import OrderItem from "..";

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("<OrderItem />", () => {
  it("should render correctly", async () => {
    const { container } = render(
      <BrowserRouter>
        <OrderItem
          orderId="32323"
          createdAt="2021-03-10T09:48:02.653Z"
          state={2}
          onViewOrderDetail={() => {}}
          orderItems={[
            {
              sku: "2",
              price: 200000,
              imageUrl:
                "https://assets.adidas.com/images/h_840,f_auto,q_auto:sensitive,fl_lossy/55d6296fe40140468240ac9300a5bb62_9366/Crew_Sweatshirt_Multicolor_HB4763_21_model.jpg",
              name: "CREW SWEATSHIRT",
              quantity: 1,
            },
          ]}
        />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
