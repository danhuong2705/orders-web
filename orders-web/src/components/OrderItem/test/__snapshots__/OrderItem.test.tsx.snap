// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItem /> should render correctly 1`] = `
<div
  class="css-1ldu4iu"
>
  <div
    class="css-0"
  >
    2021-03-10
  </div>
  <div
    class="css-1dlse9"
  >
    CREW SWEATSHIRT
     
     
  </div>
  <div
    class="css-1kewfy7"
  >
    DELIVERED
  </div>
</div>
`;
